Runtime/NativeVoiceChatClient.cs

```diff
@@ -0,0 +1,186 @@
+
+        private readonly AudioClip mic;
+
+        private bool mute;
+
+        /// <summary>
+        /// Creates NativeVoiceChatClient with peerClient and voiceChatConfig.
+        /// </summary>
+        /// <param name="peerClient">Peer client.</param>
+        /// <param name="voiceChatConfig">Voice chat config.</param>
+        public NativeVoiceChatClient(
+            NativePeerClient peerClient, VoiceChatConfig voiceChatConfig)
+        {
+            voiceChatContainer = new GameObject("VoiceChatContainer").transform;
+            Object.DontDestroyOnLoad(voiceChatContainer);
+
+            resources = new Dictionary<string, (
+                NativeInOutAudio inOutAudio, MediaStream inStream,
+                AudioStreamTrack inTrack, MediaStream outStream)>();
+            mute = voiceChatConfig.InitialMute;
+            peerClient.AddPcCreateHook(CreatePc);
+            peerClient.AddPcCloseHook(ClosePc);
+
+            mic = Microphone.Start(null, true, 1, 48000);
+            while (!(Microphone.GetPosition(null) > 0))
+            {
+                // do nothing
+            }
+        }
+
+        private void CreatePc(string id, bool isOffer, RTCPeerConnection pc)
+        {
+            if (resources.ContainsKey(id))
+            {
+                // Not covered by testing due to defensive implementation
+                return;
+            }
+
+            var inOutAudio = GetInOutAudio();
+
+            var inTrack = new AudioStreamTrack(inOutAudio.InAudio)
+            {
+                Loopback = false
+            };
+            var inStream = new MediaStream();
+            pc.AddTrack(inTrack, inStream);
+            if (Logger.IsDebug())
+            {
+                Logger.LogDebug($"AddTrack(IN): id={id}");
+            }
+
+            var outStream = new MediaStream();
+            outStream.OnAddTrack += evt =>
+            {
+                if (Logger.IsDebug())
+                {
+                    Logger.LogDebug($"OnAddTrack(OUT): kind={evt.Track.Kind} id={id}");
+                }
+                if (evt.Track is AudioStreamTrack outTrack)
+                {
+                    inOutAudio.OutAudio.SetTrack(outTrack);
+                }
+            };
+            pc.OnTrack += evt =>
+            {
+                if (Logger.IsDebug())
+                {
+                    Logger.LogDebug($"OnTrack(OUT): kind={evt.Track.Kind} id={id}");
+                }
+                if (evt.Track.Kind == TrackKind.Audio)
+                {
+                    outStream.AddTrack(evt.Track);
+                }
+            };
+
+            resources.Add(id, (inOutAudio, inStream, inTrack, outStream));
+        }
+
+        private NativeInOutAudio GetInOutAudio()
+        {
+            var inOutAudioGo = new GameObject("InOutAudio");
+            inOutAudioGo.transform.SetParent(voiceChatContainer);
+            var inOutAudio = inOutAudioGo.AddComponent<NativeInOutAudio>();
```

Comment:
ここだけ他 2 つと処理の順番が異なるのが気になります。

```suggestion
            var inOutAudioGo = new GameObject("InOutAudio");
            var inOutAudio = inOutAudioGo.AddComponent<NativeInOutAudio>();
            inOutAudioGo.transform.SetParent(voiceChatContainer);
```

Reply:
fixed https://github.com/extreal-dev/Extreal.Integration.Chat.WebRTC/pull/1/commits/0a91e92fb544cac18e0010fd0e69a4b533478ec1

Reply:
確認しました！